pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1360,10 +1360,13 @@ public ShutdownService getShutdownService() {
         return shutdownService;
     }
 
+    public static String advertisedAddress(ServiceConfiguration config) {
```

Comment:
What is the difference of this method compared to `org.apache.pulsar.broker.ServiceConfigurationUtils.getAppliedAdvertisedAddress` added in #10312 ? 

Would it be possible to add javadoc with the explanation?

Reply:
done

Reply:
@315157973 Great. It is still a bit unclear to me what is the difference of this method compared to org.apache.pulsar.broker.ServiceConfigurationUtils.getAppliedAdvertisedAddress added in #10312 . Do you have an answer for that question?

Reply:
The order of getAppliedAdvertisedAddress() to obtains addresses is:
AdvertisedAddress, advertisedListener, hostname
The order of advertisedAddress() to obtain addresses is:
AdvertisedAddress, hostname

Why the advertisedAddress() method does not obtain the advertisedListener has been described in PR

Reply:
> Why the advertisedAddress() method does not obtain the advertisedListener has been described in PR

It's not clear to me how it is described. Could you elaborate?

Reply:
You can think that it just rolled back some code https://github.com/apache/pulsar/pull/10312
Because brokerUrl() can’t get the port based on advertisedListener, advertisedListener is a Map, each value has a port, and we don’t know what the key is
We set `brokerServicePort` and `advertisedAddress` in broker.conf, where the address should be advertisedAddress:brokerServicePort, if `advertisedAddress` is not set, it should be the hostname of the machine, namely: hostname:brokerServicePort

Reply:
The doc is here, PTAL
https://pulsar.apache.org/docs/en/concepts-multiple-advertised-listeners/

Reply:
The address now obtained is `advertisedListenerAddress:brokerServicePort.`
The address we expect should be `advertisedListenerAddress:advertisedListenerPort`
But we don't know which advertisedListenerAddress should be used, because we don‘t know the key. 
Therefore, we can only get advertisedAddress and hostname

Reply:
Thanks for clarifying it @315157973 . However, this isn't clarified in the code.

The original problem remains: there's the `PulsarService.advertisedAddress` method which is almost the same as `org.apache.pulsar.broker.ServiceConfigurationUtils.getAppliedAdvertisedAddress` , but there's no description in the code why these are different and why there's a need to have 2 separate methods. 

It would be good to summarize this conversation in the javadocs & code comments since that's what maintainers of the code will be reading.
Perhaps @Anonymitaet could help in documenting this?

Reply:
@lhotari thanks for the suggestion
Since @315157973 is the developer of this and can provide technically correct explanations, would you like to add descriptions in javadoc? Then I would like to review, thanks

Reply:
This API will not be exposed to users, only exists with Broker. I will add comments to the method.

Reply:
PTAL. Do you have any suggestions for the comments?

Reply:
The difference between `PulsarService.advertisedAddress` and `org.apache.pulsar.broker.ServiceConfigurationUtils.getAppliedAdvertisedAddress` hasn't been addressed in the code comments. That was my original feedback. 

It seems wrong to have duplication in code between PulsarService and ServiceConfigurationUtils without a clear distinction. There would be no need to clarify this duplication if there's a way to eliminate the duplication. Do you have a way to resolve the duplication with further refactoring?

Reply:
How about add such an method in `ServiceConfigurationUtils`?

getAppliedAdvertisedAddress(ServiceConfiguration configuration, boolean ignoreAdvertisedListener)

Reply:
> How about add such an method in `ServiceConfigurationUtils`?
> 
> getAppliedAdvertisedAddress(ServiceConfiguration configuration, boolean ignoreAdvertisedListener)

:+1:  I think this could resolve the issue. Documenting the reason and meaning of `ignoreAdvertisedListener` in javadoc would be helpful.

Reply:
@315157973 pls ping me if you need doc review